core/integration/ref_test.go

```diff
@@ -0,0 +1,44 @@
+package core
+
+import (
+	"strings"
+	"testing"
+
+	"dagger.io/dagger"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRefIntegration(t *testing.T) {
```

Comment:
We should also have some more complete integration tests - see https://github.com/dagger/dagger/blob/9d8f5cb804f8d647191b3f803f846a23a1b336ad/core/integration/module_config_test.go#L1169-L1185

We can mirror https://github.com/dagger/dagger-test-modules to some places in gitlab, and parameterize the tests to run against that repo hosted in different places (like a subgroup in gitlab). That way, we're testing against these other VCSs in the same way that we're testing against github, so they're both first-class citizens.

Reply:
I discussed it with @marcosnils and I am not sure that integration tests are necessary here. Most of the logic here is to basically split the ref into root of repo and subdir. With the tests on vcs. repoRootForImportPath and the new `vcs.ExtractRootAndSubdirFromRef`

Reply:
Will re-review in a bit - but. I don't necessarily think we *need* to test this for an initial merge, but we absolutely need some sanity integration tests at *some* point soon.

Previously, changes to some logic in git resolution/etc changes this in *incredibly* subtle and unexpected ways. If we aren't testing this, this becomes much more likely. This *has* happened before, multiple times - the module source logic gets touched a lot, and from experience it seems to be very fragile.

Reply:
Ok, this makes sense 👌. I was planning on adding integration tests for the private git support, it seems a great complementary add at this time, wdyt?

Reply:
SGTM :rocket: